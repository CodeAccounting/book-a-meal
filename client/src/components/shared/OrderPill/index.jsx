import React from 'react';
import moment from 'moment';
import className from 'classnames';
import { Link } from 'react-router-dom';
import { calculateCashEarnedFromOrder } from '../../../helpers';
import { catererOrderObjPropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @function OrderPill
 * @returns {JSX} OrderPill
 */
const OrderPill = ({ order }) => {
  const statusClass = className({
    danger: order.status === 'canceled',
    warning: order.status === 'pending',
    success: order.status === 'delivered',
  });

  return (
    <div className="order-history-pill admin-order-history-pill">
      <Link href={`/orders/${order.id}`} to={`/orders/${order.id}`}>
        <div className="order-history-header">
          <h3>{`#${order.id}`}</h3>
          <h3>&nbsp;&nbsp;{moment(order.createdAt).format('YY/MM/DD hh:mm')}</h3>
        </div>
        <hr />
        <div>
          <p>{`${order.customer.firstname} ${order.customer.lastname}`}&nbsp; - &nbsp;{`${order.deliveryAddress}`}&nbsp; - &nbsp;{`${order.deliveryPhoneNo}`}</p>
        </div>
        <div className="order-history-footer">
          <h4>Status:
            <span style={{ textTransform: 'capitalize' }} className={statusClass}> {order.status}</span>
          </h4>
          <h2>&#8358;{calculateCashEarnedFromOrder(order.meals)}</h2>
        </div>
      </Link>
    </div>
  );
};

OrderPill.propTypes = {
  order: catererOrderObjPropTypes.isRequired
};

export default OrderPill;
