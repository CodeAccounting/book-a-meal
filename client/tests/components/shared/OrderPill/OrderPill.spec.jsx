import React from 'react';
import OrderPill from '../../../../src/components/shared/OrderPill';
import { caterersOrdersObj, customersOrdersObj, caterer, customer } from '../../../setup/mockData';


describe('OrderPill', () => {
  it('renders Caterer Order correctly', () => {
    const wrapper = shallow(<OrderPill
      order={caterersOrdersObj.orders[0]}
      user={caterer}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Customer Order correctly', () => {
    const wrapper = shallow(<OrderPill
      order={customersOrdersObj.orders[0]}
      user={customer}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Customer Order correctly and adds ... for more than one meal', () => {
    const wrapper = shallow(<OrderPill
      order={customersOrdersObj.orders[1]}
      user={customer}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('p').text()).toEqual('1x Vegetable Sharwama and Guava Smoothie...');
  });

  it('renders Customer Order correctly when status started', () => {
    const wrapper = shallow(<OrderPill
      order={customersOrdersObj.orders[3]}
      user={customer}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('h4').text()).toEqual('Status: Pending');
  });
});
