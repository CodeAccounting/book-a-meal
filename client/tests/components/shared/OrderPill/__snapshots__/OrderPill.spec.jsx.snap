// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderPill renders correctly 1`] = `
<div
  className="order-history-pill admin-order-history-pill"
>
  <Link
    href="/orders/fb097bde-5959-45ff-8e21-51184fa60c25"
    replace={false}
    to="/orders/fb097bde-5959-45ff-8e21-51184fa60c25"
  >
    <div
      className="order-history-header"
    >
      <h3>
        #fb097bde-5959-45ff-8e21-51184fa60c25
      </h3>
      <h3>
          
        18/04/06 12:47
      </h3>
    </div>
    <hr />
    <div>
      <p>
        Emiola Olasanmi
          -  
        4, Church Street, Yaba
          -  
        +2348134567890
      </p>
    </div>
    <div
      className="order-history-footer"
    >
      <h4>
        Status:
        <span
          className="success"
          style={
            Object {
              "textTransform": "capitalize",
            }
          }
        >
           
          delivered
        </span>
      </h4>
      <h2>
        ₦
        1200
      </h2>
    </div>
  </Link>
</div>
`;
